tests/cuda/test_cuda_main.cu

```diff
@@ -0,0 +1,21 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include "../../cuda/src/utils/cuda_error_handling.hpp"
+
+int main(int argc, char** argv) {
+    testing::InitGoogleTest(&argc, argv);
+
+    int r = RUN_ALL_TESTS();
+
+    VECMEM_CUDA_ERROR_CHECK(cudaDeviceReset());
```

Comment:
As you saw from the CI, this will have to be made a bit more elaborate... In the past I did things like:

```c++
int nDevices = 0;
cudaGetDeviceCount(&nDevices);
if(nDevices > 0) {
   VECMEM_CUDA_ERROR_CHECK(cudaDeviceReset());
}
```

Notice that the return value of `cudaGetDeviceCount(...)` is not checked. Since running that function on a system without the NVidia driver loaded will result in an error code return. It will however silently leave `nDevices` at `0` in that case.

Also: this file should be called `test_cuda_main.cpp`. Let's only use `.cu` for the files that can only be processed by `nvcc`. (I.e. mostly just files with kernel declarations and launches.)